### PR TITLE
Add basic UTO unit-test

### PR DESCRIPTION
**Pull request by utkarsharma2**
# Description
## What is the current behavior?
Currently, the basic framework for unit-test cases is missing from UTO

related: #1724 

## What is the new behavior?
Added basic setup for unit-test cases, so that other developers can leverage it.

## Does this introduce a breaking change?
Nope

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: universal_transfer_operator/test/test_data_provider/test_data_provider.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+@pytest.mark.parametrize(
+    "datasets",
+    [
+        {"dataset": File("s3://astro-sdk-test/uto/", conn_id="aws_default"), "expected": S3DataProvider},
+        {"dataset": File("gs://uto-test/uto/", conn_id="google_cloud_default"), "expected": GCSDataProvider},
+    ],
+    ids=lambda d: d["dataset"].conn_id,
+)
+def test_create_dataprovider(datasets):
```
Comment (sunank200) [inline universal_transfer_operator/test/test_data_provider/test_data_provider.py:17]: Doc-string would be useful.

**Review by sunank200 — COMMENTED** [review]
Looks good. We can also add CI changes to run the test in this PR.

**Comment by utkarsharma2** [comment]
> Looks good. We can also add CI changes to run the test in this PR.

I'll add a separate PR for CI, meanwhile, you can also add few tests.

File: universal_transfer_operator/test/test_data_provider/test_data_provider.py (hunk 1 above)
Comment (utkarsharma2) [reply universal_transfer_operator/test/test_data_provider/test_data_provider.py:17]: Done.

**Review by sunank200 — APPROVED** [review]
It should be okay to merge once deepsource is fixed

**Pull request merged**